packages/server/src/unstable-core-do-not-import/contentTypeParsers.ts

```diff
@@ -0,0 +1,31 @@
+import { Readable } from 'stream';
+import { TRPCError } from './error/TRPCError';
+import type { ParserZodEsque } from './parser';
+
+export type UtilityParser<TInput, TOutput> = ParserZodEsque<TInput, TOutput> & {
+  parse: (input: unknown) => TOutput;
+};
+
+// Should be the same possible types as packages/client/src/links/internals/contentTypes.ts isOctetType
+
+/**
+ * When expecting a supported octet type to be passed from the frontend, this parser may be used to validate the type for your procedure
+ */
+export function parseOctetInput<
+  TInput extends File | Blob | Uint8Array,
```

Comment:
```suggestion
  TInput extends Blob | Uint8Array,
```

Reply:
Hm this one's trickier, thing is while the TS type is on the backend, it propagates to the frontend's types and is essential if you want to send a file.

Should we do a little ponyfill somehow? Not sure how to approach that